### PR TITLE
feat(estimation): mixture SEs/covariance, p(k) form, .fitrx round-trip [#983]

### PR DESCRIPTION
**Pull request by roninsightrx**
## Why

Follow-up to #977 / #980 (mixture models). #983 tracks what the Phase 1–5 work
deliberately left out. This PR lands three of those items — the highest-priority
one (standard errors) plus two bounded ones — and a batch of correctness fixes
found along the way:

- **Item 1 — SEs / covariance (highest priority).** Mixture fits skipped the
  covariance step entirely: no SEs, RSEs, or covariance matrix.
- **Item 4 — `p(k)` mixing form.** Already evaluated by the objective but untested
  and under-documented.
- **Item 5 — `.fitrx` checkpoint.** A restored mixture fit dropped its
  `MIXEST`/`PMIX_*` per-subject columns.

Items 2 (SAEM / IMP / Bayes) and 3 (IOV under a mixture) remain open on #983.

## What changed

**Standard errors / covariance (item 1).**
- `compute_covariance` now branches on `template.mixture`: its finite-difference
  Hessian is built on the K-fold mixture objective
  (`estimation::mixture::mixture_ofv`, already on the −2·logL scale, reconverging
  every (subject × class) inner solve), not the single-population marginal. The
  single-population reconverge / analytic-R-matrix paths are single-population-only
  and are bypassed for mixtures; both the FD-step-selection and Hessian-stencil
  objectives route through one shared `cov_ofv` closure.
- Removed the two mixture bypass guards in `outer_optimizer` (eval-only and NLopt
  finalizer), so the step runs for mixtures like any other fit.
- `packed_param_label` names the per-class Ω/Σ override coordinates
  (`omega[<eta>_MIX{k}]` / `sigma[<sigma>_MIX{k}]`) instead of a generic
  `packed[N]`. The mixing-logit coefficients are ordinary thetas and already label
  by name.
- **SE convention for the mixing fraction:** reported on the scale the mixing form
  is parameterized in — `logit(k)` coefficients on the logit scale, `p(k)` on the
  probability scale directly.

**`p(k)` mixing form (item 4).** The direct-probability form was already evaluated
(probabilities used directly, then renormalised); this adds tests pinning it and
documents that only the softmax `logit` form has an analytic outer gradient (a
`p(k)` fit uses FD).

**`.fitrx` round-trip (item 5).** Per-subject `MIXEST` + `PMIX_1..K` persist as
optional trailing columns on `ebes.csv`, appended only for a mixture fit (a
non-mixture bundle is byte-identical to before). The reader now treats the base
columns as a minimum and locates the mixture columns by header name, so old
bundles still load.

**Correctness fixes (bundled, `bab7b0d4`).** Class-resolved outer gradient on each
rayon worker (a `MIXNUM`-branched typical value was computing class 2 on the
class-1 branch — a silently class-swapped gradient); mixing-expression covariates
registered as required data columns (were read as 0); `MIXNUM` rejected everywhere
outside a `[mixture]` block; eval-only runs emit `PMIX_*`/`MIXEST`.

## Alternatives considered

For the covariance objective, threading a closure through `compute_covariance`
would have been more general but touches every caller; branching internally on
`template.mixture` is the smaller, lower-risk change and keeps the single shared
`cov_ofv` so the base OFV, FD-step selection, and Hessian stencil cannot drift.

## Cross-repo dependency
| Repo | PR | Status | Must merge |
|------|----|--------|------------|
| ferx (R) | FeRx-NLME/ferx-r#___ | not needed | — |

No public Rust API signatures changed (all new behaviour is internal or additive
to existing `FitResult` fields already parsed by ferx-r).

## Breaking changes
- [x] None

## Numerical validation

New NONMEM anchor `tests/nonmem/mixture_iv_cov.ctl` = the constant-mixing anchor
plus `$COVARIANCE MATRIX=R`. `MATRIX=R` is the pure Hessian-inverse `R⁻¹` ferx
computes (NONMEM's default sandwich `R⁻¹SR⁻¹` differs by the usual 10–25% between
the two estimators). ferx SEs agree to **< 1 %** on all four estimated parameters:

```
SE          NONMEM (MATRIX=R)   ferx        rel
TVCL1       0.105642            0.106674    0.98%
TVCL2       0.254347            0.255942    0.63%
V           0.186107            0.185986    0.065%
p(1)        0.102569            0.102960    0.38%   (delta-method: SE(p)=p(1-p)·SE(logit))
```

- [x] Compared against: NONMEM 7.5.1 `$COVARIANCE MATRIX=R`
- [x] Reference values pinned in the slow-gated `tests/mixture_nonmem.rs`

## Performance
- [x] No significant impact expected (covariance step is opt-in / post-convergence;
      non-mixture and non-covariance paths unchanged).

## Tests
- [x] Unit test(s) added (`cargo test --lib` passes — 2880 lib tests green)
- [x] Regression tests added (each fails without its fix):
  - `mixture_covariance_step_runs_and_sizes`, `packed_param_label_names_mixture_overrides`
  - `mixture_pform_matches_equivalent_logit`, `mixture_gradient_pform_falls_back_to_fd`
  - `roundtrip_mixture_pmix_mixest`, `roundtrip_non_mixture_has_no_pmix`
  - `mixture_fit_matches_nonmem` now asserts SEs (slow-gated)

## Example (user-facing features only)
- [x] Not applicable (SEs/checkpoint are additive to the existing mixture feature;
      `p(k)` syntax already documented, now with a worked equivalence).

## Docs
- [x] `docs/estimation/mixture.qmd` updated (SE section + NONMEM SE table; `p(k)`
      vs `logit(k)` form note)
- [x] `CHANGELOG.md` `[Unreleased]` entries added

## Checklist
- [x] `cargo clippy` clean (no new warnings in changed lines)
- [x] `Dual2` sensitivity path untouched by the covariance change (FD-only)
- [x] No version bump (non-breaking)

## Reviewer hints

- The heart of item 1 is `compute_covariance`'s `is_mixture` branch + the shared
  `cov_ofv` closure (`src/estimation/covariance.rs`). Confirm the mixture OFV is
  already on the −2·logL scale (it is — no ×2, unlike the `pop_nll` path).
- `packed_param_label`'s new branch mirrors `parameterization::coordinate_names`
  exactly; the `mixture_override_len` bound keeps a truly out-of-range index on
  the `packed[N]` fallback.
- `.fitrx`: the reader change makes `ebes.csv`'s base column count a *minimum* and
  finds mixture columns by name — old bundles (no MIXEST/PMIX) still load.

## Open questions

- Structural SEs match NONMEM's `MATRIX=R` (pure `R⁻¹`), not its default sandwich.
  That is the correct apples-to-apples comparison for what ferx computes, but if a
  sandwich estimator is wanted for mixtures it would be a separate change.

### CODE REVIEW

**Comment by roninsightrx** [comment]
## Code review — mixture SEs/covariance, p(k) form, .fitrx round-trip

Core numerical machinery is solid: analytic posterior-weighted outer gradient, pack/unpack of the per-class override tail, softmax / p-form probabilities, PMIX/MIXEST output and `.fitrx` round-trip — all covered by Dual2-vs-FD parity and round-trip tests. Four findings, most severe first.

### 1. `cov_inner_tol` is silently a no-op for mixture SEs — `src/estimation/mixture.rs:135` (via `covariance.rs:685`)

The covariance step branches to `cov_ofv` → `mixture_ofv`, which reconverges every (subject × class) EBE at `options.inner_tol`, **not** the covariance step's `cov_inner_tol`. A user who fits loosely (`inner_tol = 1e-3`) then sets `cov_inner_tol = 1e-10` for trustworthy SEs gets a Hessian built on under-converged EBEs at every FD-stencil point, with no diagnostic. This is exactly the silent contamination the single-population path documents and deliberately avoids (`covariance.rs:663`, reconverge at `cov_inner_tol`). The mixture branch reintroduces it. **Strongest finding.**

### 2. Non-finite base-OFV diagnostic inspects only the base (class-1) Omega — `src/estimation/covariance.rs:739`

When a per-class `omega(2)` / `sigma(2)` override collapses and makes `cov_ofv(x_hat)` non-finite, the diagnostic re-unpacks `params_at` and checks only `params_at.omega.matrix` (base / class-1), which is well-conditioned. It then reports "numerical overflow/underflow in model evaluation" instead of "Omega not positive definite", pointing the user at the wrong cause.

### 3. Explicit built-in optimizer silently downgraded to BOBYQA — `src/estimation/outer_optimizer.rs:137`

For a mixture, a user-chosen `bfgs` / `trust-region` / `gn` optimizer falls through the resolved-optimizer match to `Optimizer::Bobyqa` and runs — but unlike the flat-theta preflight and other fallbacks, nothing is appended to `FitResult.warnings`. The user's explicit choice is dropped invisibly with no indication a different optimizer executed.

### 4. Per-class EBE non-convergence under-reported — `src/estimation/mixture.rs:173`

Convergence / fallback / hard-reject stats are recorded only for each subject's **winning** class. In a K=2 fit where subject *i*'s class-2 EBE fails but class 1 wins the posterior, `converged[best][i]` reads only class 1, so `n_unconverged` stays 0. The class-2 `nll_i2` still enters the log-sum-exp OFV with weight `p_i2`, so a contaminated likelihood contribution passes the `optimize_nlopt` EBE-convergence guard with no user-facing warning.

---
Verified non-issues: `mu_k` is ignored inside `find_ebe` (`let _ = mu_k;`), so its omission on the mixture path is harmless; `find_ebe` is serial, so the `MIXTURE_CLASS` thread-local is safe.

_Reviewed at effort: medium._

File: src/io/fitrx.rs [hunk 1]
```diff
@@ -983,26 +1016,39 @@ fn parse_subjects(
...
+            header_cols, base_cols
         )));
     }
+    // Optional mixture columns (#983): MIXEST + PMIX_1..K, by header name.
+    let mixest_idx = header_names.iter().position(|&c| c == "MIXEST");
+    let pmix_idxs: Vec<usize> = header_names
+        .iter()
+        .enumerate()
+        .filter(|(_, c)| c.starts_with("PMIX_"))
+        .map(|(i, _)| i)
```
[the remark this answers is not in the document]
Comment (roninsightrx) [reply src/io/fitrx.rs]: Good catch — fixed. `PMIX_*` columns are now restored in class-number order via a `pmix_column_order()` helper that parses the numeric suffix and sorts by it, so an alphabetically-reordered header (`PMIX_10` before `PMIX_2` at K ≥ 10) maps each column to the right class instead of silently swapping probabilities. A non-numeric suffix sorts last, deterministically. Covered by a new unit test `pmix_column_order_sorts_by_class_number` (identity for an in-order header, correct remap for a scrambled one).

**Comment by roninsightrx** [comment]
## Review findings addressed — pushed in e2c70ef6

All four findings from the code review above, plus Copilot's `.fitrx` ordering finding, are fixed. Each carries a regression test.

| # | Fix | Test |
|---|-----|------|
| 1 | Covariance step reconverges per-class EBEs at `cov_inner_tol`, not the fit's `inner_tol` (own override-clone of options) — a loose fit + tight `cov_inner_tol` for trustworthy SEs is now honoured instead of silently building the Hessian on loose EBEs. | `mixture_covariance_uses_cov_inner_tol_not_fit_tol` |
| 2 | Non-finite base-OFV diagnostic inspects the worst-conditioned class Omega (new `diagnostic_omega`), so a collapsed per-class `omega(k)`/`sigma(k)` override is reported as Omega collapse, not a model-eval overflow. | `diagnostic_omega_picks_collapsed_class_override` |
| 3 | An explicitly chosen optimizer that can't drive a mixture (built-in BFGS/L-BFGS, trust-region, GN) still runs under BOBYQA but now warns to `FitResult.warnings`; `auto` stays silent by design. | `mixture_downgraded_optimizer_warns`, `mixture_default_optimizer_does_not_warn` |
| 4 | EBE non-convergence / fallback / hard-reject counted over every *contributing* class (new `any_contributing`, weight > 1e-6), not just the winning class — a non-converged non-winning class is no longer hidden from the outer convergence guard. | `any_contributing_scores_nonwinning_classes` |
| 5 (Copilot) | `.fitrx` restore orders `PMIX_*` columns by class number (new `pmix_column_order`), so a reordered header (`PMIX_10` before `PMIX_2`, K ≥ 10) no longer silently swaps class probabilities. | `pmix_column_order_sorts_by_class_number` |

Full lib suite green (2886 passed), rustfmt clean, no new clippy warnings. Changelog updated.
